server/src/main/java/io/crate/execution/dml/upsert/TransportShardUpsertAction.java

```diff
@@ -231,6 +236,9 @@ protected WritePrimaryResult<ShardUpsertRequest, ShardResponse> processRequestIt
                         item.id(),
                         getExceptionMessage(e),
                         (e instanceof VersionConflictEngineException)));
+            } catch (AssertionError e) {
```

Comment:
this is only for tests, no? should we maybe have a comment to make it clear?

Reply:
Added a comment